### DISCUSSION TITLE
I need help for custom authentication using JWT.

### DISCUSSION THREAD

**Discussion by GSaphal** [Web3Auth/Web3Auth#1246] [Help]
I need your help to figure out an issue. Currently, we are trying to integrate Custom Authentication using our own JWT generated by our backend. 

What did we till now?
We generated a JWT token based on our own JWKs, and the token is getting decoded.

What is the issue right now?
When we try to log in using jwt from the web3 auth core package connectTo function, we get this error: 

`Unable to detect login share from the Auth Network. This may be due to slow internet connection. Check your internet speed and try again. If you're using a vpn, please turn it off.`

I know there is something with the configuration but still need help figuring it out.

Our Configuration Code:

 ```
 const initializeWeb3Auth = async () => {
    console.log('Initialize Web3Auth');
    setIsLoading(true);
    const web3Object = new Web3AuthCore({
      clientId:
      chainConfig: currentChainConfig,
    });

    if (adapterName === 'openlogin') {
      const network=testnet;
      const openloginAdapter = new OpenloginAdapter({
        adapterSettings: {
          network,
          uxMode: 'redirect',
          loginConfig: {
            jwt: {
              name: 'DUMMY_NAME',
              verifier: 'evolvtestnet',
              typeOfLogin: 'jwt',
            },
          },
      });
      web3Object.configureAdapter(openloginAdapter);
      console.log('After Configuration', web3Object);
      if (web3Object.status === 'connected') {
        setProvider(web3Object.provider);
      }
      await web3Object.init();
      console.log('Initialized Web3 Object', web3Object);
      setWeb3Auth(web3Object);
      setProvider(web3Object.provider);
    }
  };
```

Connect Function:
  ```
web3Auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
            loginProvider: 'jwt',
            extraLoginOptions: {
              id_token: '', // This is generated by our Backend.
              verifierIdField: 'sub',
            },
          })
```

Example Token:
`eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjFiYjk2MDVjMzZlMWJiOTYwNWMzNmVHNi1pUDdDXzROZHBwRndpRXVja05DVV9ldEZJVkJCeHh0NFVIOWNCOUcwNjkzODY4MzAyMDJiMmQ2OTM4NjgzMDIwMmIyZCJ9.eyJzdWIiOiIiLCJuYW1lIjoiTW9oYW1tYWQgU2hhaGJheiBBbGFtIiwiZW1haWwiOiJzaGFoYmF6QHdlYjNhdXRoLmlvIiwiYXVkIjoidXJuOnNhcGhhbHRlc3RuZXQiLCJpc3MiOiJodHRwczovL2Rldi5kaWdpdGFsc2FtYWRoYW5uZXBhbC5jb20iLCJpYXQiOjE2NzQ1MzQ0MzAsImV4cCI6MTY3NDUzODAzMH0.ns8F5kIWr_muoU0bKU4jtT1ilCjPDLXE_2zdQCi3Glox2JQIL8tAdUI5Vv2HVpZlDy7uVrvFi34gTN2xGmOkdYEsklGonLYBMIIK_4dWapEbKFd2wMQsQqbjz7MZNpOxLNul1Cleyeaxo4vfyJ6YYbmU7sf71NM7ShDyri2PE563Q7MF6MjUinvR3dcWnt16oBnoI7dxc1LYIiWM7XgtQFNiNAZOhTDd7jGid_SGHLgfvw8ld_uXMV5vhKyttPMbafr45y789KqGYLMuf_etHfXck9Qk0Q6fLowF5unJO1d5wYp6XmdfhVdcO4VhKi8ascUaXK3UEuk_c55Hqw`

**Comment by snaka** [comment]
@GSaphal 
I am interested in your problem and would like to understand it.
However, the formatting of the code you have provided seems to be broken, making it difficult to understand the problem itself.

If you could please format the code, someone may be able to comment on your problem.

**Reply by GSaphal** [reply]
Sorry for that. I have formatted the code.

**Comment by arch1995** [comment]
Hi @GSaphal 

We tried decoding your `jwt` token and found out that `sub` is empty.

<img width="772" alt="image" src="https://user-images.githubusercontent.com/29117472/215031599-1345e203-68c6-44e5-8001-88f283212e71.png">

Please provide the sub when you are generating a new `idtoken` and it has to be unique for each user.

**Reply by GSaphal** [reply]
Thanks a lot, @arch1995. Adding a unique sub while generating tokens resolved the issue.
